e3sm_diags/derivations/formulas_cosp.py

```diff
@@ -0,0 +1,435 @@
+#     if missing. FIXME: These default_bnds might not align with the number of
+#     cloud axis coordinates in a dataset.
+# 3. 'min_mask': the minimum value to standardize the axis on for histogram
+#     plotting.
+CloudAxis = Literal["prs", "tau"]
+CloudHistMapAttrs = TypedDict(
+    "CloudHistMapAttrs",
+    {"keys": List[str], "default_bnds": np.ndarray, "min_mask": float},
+)
+CLOUD_HIST_MAP: Dict[CloudAxis, CloudHistMapAttrs] = {
+    "prs": {
+        "keys": ["cosp_prs", "cosp_htmisr", "modis_prs", "misr_cth", "isccp_prs"],
+        "default_bnds": np.array(
+            [
+                [1000.0, 800.0],
+                [800.0, 680.0],
+                [680.0, 560.0],
+                [560.0, 440.0],
+                [440.0, 310.0],
+                [310.0, 180.0],
+                [180.0, 0.0],
+            ]
+        ),
+        "min_mask": 0.0,
+    },
+    "tau": {
+        "keys": ["cosp_tau", "cosp_tau_modis", "modis_tau", "misr_tau", "isccp_tau"],
+        "default_bnds": np.array(
+            [
+                [0.3, 1.3],
+                [1.3, 3.6],
+                [3.6, 9.4],
+                [9.4, 23],
+                [23, 60],
+                [60, 379],
+            ]
+        ),
+        "min_mask": 0.3,
+    },
+}
+
+
+# A dictionary mapping the target variable key to the "prs" and "tau" axes
+# adjustment ranges. If either value in the (min, max) tuple is None, then
+# the actual value from the axis is used instead.
+AdjRange = Tuple[Optional[float], Optional[float]]
+CLOUD_BIN_SUM_MAP: Dict[str, Dict[str, AdjRange]] = {
+    # ISSCCP
+    "CLDTOT_TAU1.3_ISCCP": {"prs": (None, None), "tau": (1.3, None)},
+    "CLDTOT_TAU1.3_9.4_ISCCP": {"prs": (None, None), "tau": (1.3, 9.4)},
+    "CLDTOT_TAU9.4_ISCCP": {"prs": (None, None), "tau": (9.4, None)},
+    # MODIS
+    "CLDTOT_TAU1.3_MODIS": {"prs": (None, None), "tau": (1.3, None)},
+    "CLDTOT_TAU1.3_9.4_MODIS": {"prs": (None, None), "tau": (1.3, 9.4)},
+    "CLDTOT_TAU9.4_MODIS": {"prs": (None, None), "tau": (9.4, None)},
+    "CLDHGH_TAU1.3_MODIS": {"prs": (440, 0), "tau": (1.3, None)},
+    "CLDHGH_TAU1.3_9.4_MODIS": {"prs": (440, 0), "tau": (1.3, 9.4)},
+    "CLDHGH_TAU9.4_MODIS": {"prs": (440, 0), "tau": (9.4, None)},
+    # MISR
+    "CLDTOT_TAU1.3_MISR": {"prs": (None, None), "tau": (1.3, None)},
+    "CLDTOT_TAU1.3_9.4_MISR": {"prs": (None, None), "tau": (1.3, 9.4)},
+    "CLDTOT_TAU9.4_MISR": {"prs": (None, None), "tau": (9.4, None)},
+    "CLDLOW_TAU1.3_MISR": {"prs": (0, 3), "tau": (1.3, None)},
+    "CLDLOW_TAU1.3_9.4_MISR": {"prs": (0, 3), "tau": (1.3, 9.4)},
+    "CLDLOW_TAU9.4_MISR": {"prs": (0, 3), "tau": (9.4, None)},
+}
+
+# A dictionary mapping the names of "prs" axes to the unit adjustment value.
+# - COSP v2 "cosp_pr" is in units "Pa" instead of "hPa" (v1).
+# - COSP v2 "cosp_htmisr" is in units "m" instead of "km" (v1).
+PRS_UNIT_ADJ_MAP = {"cosp_prs": 100, "cosp_htmisr": 1000}
+
+
+def cosp_histogram_standardize(
```

Comment:
Maybe consider to remove this function for now.

Reply:
Are you referring to `PRS_UNIT_ADJ_MAP`?